server/handler.go

```diff
@@ -511,6 +512,24 @@ func resultForEmptyIter(ctx *sql.Context, iter sql.RowIter, resultFields []*quer
 	return &sqltypes.Result{Fields: resultFields}, nil
 }
 
+// GetDeferredProjections looks for a top-level deferred projection, marks it as deferred, and retrieves its projections
+func GetDeferredProjections(iter sql.RowIter) []sql.Expression {
```

Comment:
the alternative to this tagging `queryFlags` with a `doProjectionDefer` in the handler context and having planbuilder mark the defer for the top-level projection, which you identify with `flags.IsSet(DoProjectionDefer) && notInSubquery()` when building final projection